gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/grpc.py.j2

```diff
@@ -258,7 +258,7 @@ class {{ service.name }}GrpcTransport({{ service.name }}Transport):
             )
 
         self._grpc_channel = self._base_channel
-        if CLIENT_LOGGING_SUPPORTED and _LOGGER.isEnabledFor(std_logging.DEBUG):  # pragma: NO COVER
+        if CLIENT_LOGGING_SUPPORTED and not _LOGGER.isEnabledFor(std_logging.NOTSET):  # pragma: NO COVER
```

Comment:
(see my other comment on this code line first)

We do NOT want the `not` in  `not _LOGGER.isEnabledFor(std_logging.NOTSET)`: If this logger does not have a level specified, `isEnabledFor` will traverse up the chain until it finds a handler ([ref](https://docs.python.org/3/library/logging.html#logging.Logger.isEnabledFor)). So this use case would fail: `_LOGGER`  has no handlers (since it is `NOTSET`) but its parent is configured with, say, `DEBUG`. Then `isEnabledFor` would return `True`, the `not` will make it `False`, and the user sees no log. This runs counter to the idiomatic expectations of logging in Python.

We should remove the `not`. If checking for `NOTSET` causes an issue, then replace the argument with the value 1, which is the next highest debug level ([ref](https://docs.python.org/3/library/logging.html#levels)).

Reply:
I've reverted the statements for now but I do have a question:

> _LOGGER has no handlers (since it is NOTSET) but its parent is configured with, say, DEBUG. Then isEnabledFor would return True, the not will make it False, and the user sees no log. This runs counter to the idiomatic expectations of logging in Python.

If a level is configured against the parent, then isEnabledFor(NOTSET) should return False and the not should make it True.

The isEnabledFor should check for the level configured against all loggers up in the chain.

Reply:
https://docs.python.org/3/library/logging.html#logging.Logger.isEnabledFor

The idea for checking `not IsEnabledFor(NOTSET)` and `isEnabledFor(DEBUG)` is similar. The former imo would be better if we want to emit our logs if let's say someone has configured a logging level > DEBUG.

Reply:
> If a level is configured against the parent, then isEnabledFor(NOTSET) should return False and the not should make it True.

As per https://docs.python.org/3/library/logging.html#logging.Logger.isEnabledFor, if this logger is `NOTSET`, the check will go up the hierarchy.

> The isEnabledFor should check for the level configured against all loggers up in the chain.

Almost. Our intent is that it do so, agreed, but I noticed yesterday that the semantics are not quite that, so I filed #2292.